.init_wizard/initialize_package.py

```diff
@@ -0,0 +1,348 @@
+    exclude: Optional[List[LiteralString]] = None,
+):
+    """
+    :brief: Populate the target directory with rendered templates from the templates directory.
+    :param env: The Jinja2 environment used for rendering templates.
+    :param cfg: The configuration dictionary used to render templates.
+    :param templates_dir: The directory containing the template files.
+    :param target_dir: The directory where the rendered templates will be saved.
+    :param sub_dir: The subdirectory within the templates directory to search for templates.
+    :param exclude: A list of file patterns to exclude from processing.
+    """
+    pattern = os.path.join(templates_dir, sub_dir, "**", "*.j2")
+    absolute_files = glob(pattern, recursive=True)
+    if exclude:
+        exclude_patterns = [translate(p) for p in exclude]
+        absolute_files = [f for f in absolute_files if not any(re.search(p, f) for p in exclude_patterns)]
+    relative_files = [os.path.relpath(f, templates_dir) for f in absolute_files]
+
+    for rel_path in relative_files:
+        template = env.get_template(rel_path)
+        out_path = str(Path(rel_path)).replace(f"{sub_dir}/", "")
+        try:
+            write_to_file(f"{target_dir}/{out_path.replace('.j2', '')}", template.render(cfg))
+        except OSError as e:
+            print(f"Error writing file: {e}")
+
+
+def populate_common_files(
+    env: Environment, configuration: Dict, templates_dir: LiteralString, target_dir: LiteralString
+):
+    """
+    :brief: Populate common files in the target directory from the templates directory.
+    :param env: The Jinja2 environment used for rendering templates.
+    :param configuration: The configuration dictionary used to render templates.
+    :param templates_dir: The directory containing the template files.
+    :param target_dir: The directory where the rendered templates will be saved.
+    """
+    config_files = glob(os.path.join(templates_dir, ".*", "*.json.j2"))
+    toml = env.get_template("aica-package.toml.j2")
+    component_toml = env.get_template("aica-package.component.toml.j2")
+    controller_toml = env.get_template("aica-package.controller.toml.j2")
+
+    for cfg in config_files:
+        filepath = os.path.relpath(cfg, templates_dir)
+        template = env.get_template(filepath)
+        write_to_file(f"{target_dir}/{filepath.replace('.j2', '')}", template.render(configuration))
+
+    if len(configuration["collection_name"]) > 0:
+        cfg = {"image_name": configuration["collection_name"]}
+    elif len(configuration["controller"]) > 0:
+        cfg = {"image_name": configuration["controller"]["package_name"]}
+    else:
+        cfg = {"image_name": configuration["component"]["package_name"]}
+
+    og_delimiters = change_env_delimiters(
+        env,
+        {
+            "variable_start_string": "<<",
+            "variable_end_string": ">>",
+            "block_start_string": "<%",
+            "block_end_string": "%>",
```

Comment:
I have to do this because of the workflow replacements. Yes, I could have used this for everything but only realized at the end as .github was my last task. I can go ahead and replace the values everywhere in our .j2 files (I'm honestly looking for an argument to not do that)